plugins/formal-verification/formver.viper/src/org/jetbrains/kotlin/formver/viper/ast/Callable.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.viper.ast
+
+import scala.collection.JavaConverters
+
+/**
+ * Interface used to wrap Viper nodes with an argument list (such as function applications, method calls, ...).
+ */
+interface IntoCallable {
+    fun asCallable(): Callable
+}
+
+/**
+ * Represents a Viper-callable object with arguments.
+ */
+interface Callable {
+    val args: List<Node>
+
+    fun arg(index: Int): Node {
+        val arguments = args
```

Comment:
I'm guessing this assignment is to avoid computing `get` twice?  Perhaps `CallableImpl` should simply store the nodes?  I don't know  whether Kotlin has a good init-on-first-use idiom that would work here.  If it's a lot of trouble, this is also fine, but maybe `args` should be a function to emphasise that it is not cheap?

Reply:
> but maybe args should be a function to emphasise that it is not cheap?

That's a good idea!